### PR TITLE
[chaos-test] chaos test pipeline ingestion

### PR DESCRIPTION
**Pull request by scv119**
Run chaos test with pipeline ingestion to test its failure tolerence.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] TBD: Release tests https://console.anyscale.com/o/anyscale-internal/projects/prj_SVFGM5yBqK6DHCfLtRMryXHM/clusters/ses_6EGPcbvDzRqQghxZCr5M6RLt?command-history-section=command_history
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/chaos_test.yaml [hunk 1]
```diff
@@ -47,3 +47,19 @@
...
+  cluster:
+    app_config: dataset/pipelined_ingestion_app.yaml
+    compute_template: dataset/pipelined_ingestion_compute.yaml
+
+  run:
+    timeout: 7200
+    prepare: python wait_cluster.py 21 2400;  python setup_chaos.py
+    script: python dataset/pipelined_training.py --epochs 1 --num-windows 15  --num-files 915 --debug
+
+  stable: false
```
Comment (rkooo567) [inline release/nightly_tests/chaos_test.yaml:65]: New line!

File: release/nightly_tests/chaos_test.yaml [hunk 2]
```diff
@@ -47,3 +47,19 @@
     script: python dask_on_ray/large_scale_test.py --num_workers 150 --worker_obj_store_size_in_gb 70 --error_rate 0  --data_save_path /tmp/ray
 
   stable: false
+
+- name: chaos_pipelined_ingestion_1500_gb_15_windows
```
Comment (rkooo567) [inline release/nightly_tests/chaos_test.yaml:51]: Also update the build_pipeline.py

**Comment by scv119** [comment]
@rkooo567 looks the test is passing; but how do i tell if the chaos test is running?

**Comment by rkooo567** [comment]
@scv119 unfortunately, right now you need to manually obtain the number of killed nodes from your test script. I am planning to improve fixture API, but maybe it will be done next sprint or so. 

You can also check your driver logs to see if heartbeat failure message was printed.

```python3
node_killer = ray.get_actor("node_killer")
print(ray.get(node_killer.get_total_killed_nodes.remote()))
```

**Comment by scv119** [comment]
@rkooo567 how often do we trigger the OOM killer? looking at the logs it seems on triggered on the job started.

**Comment by rkooo567** [comment]
Looks like 

```
                sleep_interval = np.random.rand() * self.node_kill_interval_s
                time.sleep(sleep_interval)

                if node_to_kill_port is not None:
                    try:
                        self._kill_raylet(
                            node_to_kill_ip, node_to_kill_port, graceful=False)
                    except Exception:
                        pass
                    logging.info(
                        f"Killed node {node_id} at address: "
                        f"{node_to_kill_ip}, port: {node_to_kill_port}")
                    self.killed_nodes.add(node_id)
                if len(self.killed_nodes) >= self.max_nodes_to_kill:
                    break
                await asyncio.sleep(self.node_kill_interval_s - sleep_interval)
```
and node_kill_interval_s = 60 seconds.

(Looks like Stephanie added some randomness in the sleeping duration). So it could be from 0~60 seconds I think

**Comment by scv119** [comment]
oh well, we need to separate it into two PRs

**Comment by rkooo567** [comment]
> oh well, we need to separate it into two PRs

Yeah the change in test_utils might not be reflected when you run tests.

**Review by stephanie-wang — APPROVED** [review]

**Comment by stephanie-wang** [comment]
Is the plan to merge it after #20447?

**Comment by scv119** [comment]
> Is the plan to merge it after #20447?

yup i think so. the latest run passes:
https://console.anyscale.com/o/anyscale-internal/projects/prj_SVFGM5yBqK6DHCfLtRMryXHM/clusters/ses_KZRC9QBusTuQZwJBvquRiCju

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**